diff --git a/src/icon-lookup.c b/src/icon-lookup.c
index 03722db..5a50e32 100644
--- a/src/icon-lookup.c
+++ b/src/icon-lookup.c
@@ -264,18 +264,21 @@ char *find_icon_in_theme(const char *name, int theme_index, int size) {
                                 break;
                 }
                 if (match_size) {
-                        const char *suffixes[] = { ".svg", ".svgz", ".png", ".xpm", NULL };
-                        for (const char **suf = suffixes; *suf; suf++) {
-                                char *name_with_extension = g_strconcat(name, *suf, NULL);
-                                char *icon = g_build_filename(theme->location, theme->subdir_theme,
-                                                dir.name, name_with_extension,
-                                                NULL);
-                                if (is_readable_file(icon)) {
+                        const char *suffixes_1[] = { "", "-symbolic", NULL };
+                        const char *suffixes_2[] = { ".svg", ".svgz", ".png", ".xpm", NULL };
+                        for (const char **suf_1 = suffixes_1; *suf_1; suf_1++) {
+                                for (const char **suf_2 = suffixes_2; *suf_2; suf_2++) {
+                                        char *name_with_extension = g_strconcat(name, *suf_1, *suf_2, NULL);
+                                        char *icon = g_build_filename(theme->location, theme->subdir_theme,
+                                                    dir.name, name_with_extension,
+                                                    NULL);
+                                        if (is_readable_file(icon)) {
+                                                g_free(name_with_extension);
+                                                return icon;
+                                        }
                                         g_free(name_with_extension);
-                                        return icon;
+                                        g_free(icon);
                                 }
-                                g_free(name_with_extension);
-                                g_free(icon);
                         }
                 }
         }
diff --git a/src/icon.c b/src/icon.c
index 6c6baf3..0adc89d 100644
--- a/src/icon.c
+++ b/src/icon.c
@@ -222,7 +222,8 @@ char *get_path_from_icon_name(const char *iconname, int size)
         if (STR_EMPTY(iconname))
                 return NULL;
 
-        const char *suffixes[] = { ".svg", ".svgz", ".png", ".xpm", NULL };
+        const char *suffixes_1[] = { "", "-symbolic", NULL };
+        const char *suffixes_2[] = { ".svg", ".svgz", ".png", ".xpm", NULL };
         gchar *uri_path = NULL;
         char *new_name = NULL;
 
@@ -245,15 +246,19 @@ char *get_path_from_icon_name(const char *iconname, int size)
 
                         current_folder = g_strndup(start, end - start);
 
-                        for (const char **suf = suffixes; *suf; suf++) {
-                                gchar *name_with_extension = g_strconcat(iconname, *suf, NULL);
-                                maybe_icon_path = g_build_filename(current_folder, name_with_extension, NULL);
-                                if (is_readable_file(maybe_icon_path)) {
-                                        new_name = g_strdup(maybe_icon_path);
+                        for (const char **suf_1 = suffixes_1; *suf_1; suf_1++) {
+                                for (const char **suf_2 = suffixes_2; *suf_2; suf_2++) {
+                                        gchar *name_with_extension = g_strconcat(iconname, *suf_1, *suf_2, NULL);
+                                        maybe_icon_path = g_build_filename(current_folder, name_with_extension, NULL);
+                                        if (is_readable_file(maybe_icon_path)) {
+                                                new_name = g_strdup(maybe_icon_path);
+                                        }
+                                        g_free(name_with_extension);
+                                        g_free(maybe_icon_path);
+
+                                        if (new_name)
+                                                break;
                                 }
-                                g_free(name_with_extension);
-                                g_free(maybe_icon_path);
-
                                 if (new_name)
                                         break;
                         }
