diff --git a/pkgs/clan-cli/clan_cli/vars/generator.py b/pkgs/clan-cli/clan_cli/vars/generator.py
index d2c9d5e38..71ecda5ad 100644
--- a/pkgs/clan-cli/clan_cli/vars/generator.py
+++ b/pkgs/clan-cli/clan_cli/vars/generator.py
@@ -236,7 +236,7 @@ def get_machine_generators(
                             differences.append("validation_hash")
                         if differences:
                             msg = f"Machines {prev_machine} and {machine_name} have different definitions for shared generator '{gen_name}' (differ in: {', '.join(differences)})"
-                            raise ClanError(msg)
+                            log.warning(msg)
                     else:
                         shared_generators_raw[gen_name] = (
                             machine_name,
