diff --git a/pkgs/clan-cli/clan_lib/vars/generator.py b/pkgs/clan-cli/clan_lib/vars/generator.py
index 40234a04d..d3d449ded 100644
--- a/pkgs/clan-cli/clan_lib/vars/generator.py
+++ b/pkgs/clan-cli/clan_lib/vars/generator.py
@@ -269,7 +269,7 @@ def get_machine_generators(
                     )
                     if differences:
                         msg = f"Machines {prev_machine} and {machine_name} have different definitions for shared generator '{gen_name}' (differ in: {', '.join(differences)})"
-                        raise ClanError(msg)
+                        log.warning(msg)

                 else:
                     shared_generators_raw[gen_name] = (
