diff --git a/pkgs/clan-cli/clan_cli/machines/update.py b/pkgs/clan-cli/clan_cli/machines/update.py
index f9fd4010d..14bb82622 100644
--- a/pkgs/clan-cli/clan_cli/machines/update.py
+++ b/pkgs/clan-cli/clan_cli/machines/update.py
@@ -269,6 +269,7 @@ def register_update_parser(parser: argparse.ArgumentParser) -> None:
             "The machine on which to build the machine configuration.\n"
             "Pass 'localhost' to build on the local machine, or an ssh address like user@host:1234\n"
         ),
+        default=os.environ.get("CLAN_BUILD_HOST"),
     )
     add_dynamic_completer(build_host_parser, complete_build_host)
     parser.add_argument(
