diff --git a/pkgs/clan-cli/clan_lib/network/direct.py b/pkgs/clan-cli/clan_lib/network/direct.py
index 18d8395e9..1f7532982 100644
--- a/pkgs/clan-cli/clan_lib/network/direct.py
+++ b/pkgs/clan-cli/clan_lib/network/direct.py
@@ -45,7 +45,7 @@ def remote(self, peer: Peer) -> list[Remote]:
         return [
             Remote(
                 address=host,
-                user=peer.ssh_user,
+                user="jfly",
                 port=peer.port,
                 command_prefix=peer.name,
             )
diff --git a/pkgs/clan-cli/clan_lib/network/zerotier/__init__.py b/pkgs/clan-cli/clan_lib/network/zerotier/__init__.py
index d23b0c3fb..0cb66b3af 100644
--- a/pkgs/clan-cli/clan_lib/network/zerotier/__init__.py
+++ b/pkgs/clan-cli/clan_lib/network/zerotier/__init__.py
@@ -40,7 +40,7 @@ def remote(self, peer: Peer) -> list["Remote"]:
         return [
             Remote(
                 address=host,
-                user=peer.ssh_user,
+                user="jfly",
                 port=peer.port,
                 command_prefix=peer.name,
             )
