diff --git a/crates/nil/src/main.rs b/crates/nil/src/main.rs
index d9a0554..ff8baae 100644
--- a/crates/nil/src/main.rs
+++ b/crates/nil/src/main.rs
@@ -140,7 +140,7 @@ fn main_diagnostics(args: DiagnosticsArgs) {
             if max_severity > Severity::Warning {
                 process::exit(1)
             } else {
-                process::exit(0)
+                process::exit(1)
             }
         }
         Err(err) => {
