diff --git a/configure.ac b/configure.ac
index 83a27be97..632dd0a84 100755
--- a/configure.ac
+++ b/configure.ac
@@ -22,7 +22,7 @@ AC_CONFIG_MACRO_DIR([./aclocal])
 
 # Honor: Please read "Adding a new Honor-specific patch" in README.md when changing this.
 AC_INIT([thrift], [0.12.1-honor2])
-AC_DEFINE_UNQUOTED([GIT_COMMIT], ["$(git log -1 --format=format:%h)"], [Git commit hash])
+AC_DEFINE_UNQUOTED([GIT_COMMIT], ["${GIT_COMMIT:-$(git log -1 --format=format:%h)}"], [Git commit hash])
 
 AC_CONFIG_AUX_DIR([.])
 
